src/mono/mono/component/mini-wasi-debugger.c

```diff
@@ -0,0 +1,168 @@
+#include <glib.h>
+#include <mono/mini/debugger-agent-external.h>
+#include <mono/component/debugger-engine.h>
+#include <mono/metadata/components.h>
+
+#ifdef HOST_WASI
+static int conn_fd;
+static int log_level = 1;
+static int retry_receive_message = 100;
+static int connection_wait = 1000;
+
+__attribute__((import_module("wasi_snapshot_preview1")))
+__attribute__((import_name("sock_accept")))
+int sock_accept(int fd, int fdflags, int* result_ptr);
+
+static long long timeInMilliseconds()
+{
+	struct timeval tv;
+	gettimeofday(&tv,NULL);
+	return (((long long)tv.tv_sec)*1000)+(tv.tv_usec/1000);
```

Comment:
From [gettimeofday](https://c-for-dummies.com/blog/?p=4236), `tv_sec` is the number of seconds since the Unix epoch, so this function is returning the number of milliseconds since the Unix epoch.

It's possible that [clock_gettime(CLOCK_MONOTONIC, ...)](https://blog.habets.se/2010/09/gettimeofday-should-never-be-used-to-measure-time.html) would be more correct but I don't know how well support for that translates into WASI. We know at least that `gettimeofday` is working for its intended purpose in this use case.